File: Modder/modder/src/main/java/modder/ApkToolWrap.java
package modder;

// apktool stuff 

import brut.common.BrutException;
//
import java.io.File;
import java.io.IOException;

public class ApkToolWrap {

    public static void Decompile(String apkPathName, String outDirName) {
        String[] cmd = new String[]{"d", apkPathName, "--output", outDirName, "-f"};
        try {
            brut.apktool.Main.main(cmd);
        } catch (BrutException e) {
            System.out.println(e.getMessage());
        }
    }

    public static void Recompile(String decompiledFolderStr, String apkOutFileStr) {

        String[] cmd = new String[]{"b", decompiledFolderStr, "--output", apkOutFileStr};
        try {
            brut.apktool.Main.main(cmd);
        } catch (BrutException e) {
            System.out.println(e.getMessage());
        }
    }

}

File: Modder/modder/src/main/java/modder/ApkToolWrap.kt
package modder

// apktool stuff 
class ApkToolWrap {
    companion object {
        fun Decompile(apkPathName: String, outDirName: String, decodeResource: Boolean) {
            val cmd = mutableListOf("d", apkPathName, "--output", outDirName, "--force")
            // add --no-resource, because apktool by default decodes resource
            if (!decodeResource)
                cmd.add("--no-res")

            brut.apktool.Main.main(cmd.toTypedArray())
        }

        fun Recompile(decompiledFolderStr: String, apkOutFileStr: String) {

	    // https://github.com/iBotPeaches/Apktool/issues/1978
            val cmd = arrayOf("b", decompiledFolderStr, "--output", apkOutFileStr, "--use-aapt2")

            brut.apktool.Main.main(cmd)
        }
    }
}
